mccn/wcs_importer.py

```diff
@@ -0,0 +1,70 @@
+from abc import ABC, abstractmethod
+from typing import Any, List
+
+from owslib.wcs import WebCoverageService
+
+
+class WcsImporter(ABC):
+    def __init__(self, url: str) -> None:
+        self.url = url
+        # TODO: It may be poor design to instantiate a WCS in the constructor and use it later.
+        # TODO: Consider if preferable to create new instances each time a request is made.
+        self.wcs = WebCoverageService(url, version="1.0.0", timeout=300)
+
+    @abstractmethod
+    def get_metadata(self) -> None:
+        pass
+
+    @abstractmethod
+    def get_capabilities(self) -> tuple:
+        pass
+
+    @abstractmethod
+    def get_data(self, bbox: List[float]) -> Any:
+        pass
+
+
+class DemWcsImporter(WcsImporter):
```

Comment:
With just one concrete class, it's hard to judge, but it looks like this represents a one-to-one relationship between a class and a particular source (the DEM_SRTM_1Second_Hydro_Enforced product). Do you expect to create a class for each source? Shouldn't the URL be something that gets passed to an appropriate importer instance? I would think an importer class could be more abstract and reused for many sources with a similar data structure. How do you expect to map sources to importers, and what does that imply for how importers get created?

Reply:
Great question! I am planning to make use of the factory design pattern for the importers, that is one defined class per source. Since we have a finite, and in fact small, number of potential sources this is not too much overhead. When other sources are implemented I expect much common code to be moved to the base class but for now it doesn't sit there. Since all potential URLs will be accounted for this will allow us to specifically handle each one, including the particular layers or other unique features each may have.

To see how to map I have linked an [example](https://github.com/aus-plant-phenomics-network/stac-generator/blob/main/stac_generator/generator_factory.py) from the stac-generator.